rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -740,14 +742,29 @@ async fn run_controller(
         network_changes::Worker::new().context("Failed to listen for network changes")?;
 
     let mut dns_listener = network_changes::DnsListener::new()?;
+    let mut last_resolvers_sent = vec![];
 
     loop {
         tokio::select! {
             () = controller.notify_controller.notified() => {
+                let tunnel_became_ready = !controller.tunnel_ready;
+                controller.tunnel_ready = true;
                 tracing::debug!("Controller notified of new resources");
                 if let Err(error) = controller.refresh_system_tray_menu() {
                     tracing::error!(?error, "Failed to reload resource list");
                 }
+                if tunnel_became_ready {
+                    os::show_notification(
+                        "Firezone connected",
+                        "You are now signed in and able to access resources.",
+                    )?;
+                    if let Some(instant) = controller.last_connlib_start.take() {
+                        let elapsed = instant.elapsed();
+                        tracing::info!(?elapsed, "Tunnel ready");
+                    } else {
+                        tracing::error!("Impossible - tunnel is ready but we didn't record when connlib started");
```

Comment:
Do we need to cover this branch? We should use an `unreachable!` probably in that case